scheduler/server/priority_based_scheduling_alg.go

```diff
@@ -0,0 +1,230 @@
+package server
+
+import (
+	"fmt"
+	"math"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/twitter/scoot/common/stats"
+)
+
+type PriorityBasedAlg struct {
+	PriorityRatios           []int // array of priority ratios where array index == priority
+	taskModCounter           int
+	jobsByPriority           [][]*jobState // list of jobs at each priority
+	roundRobinJobIndex       []int         // for each priority, track the index of the job whose task should be assigned next
+	numBarredFromLowPriority int           // number of nodes reserved for non-lowest priority tasks
+}
+
+func MakePriorityBasedAlg(ratios []int) *PriorityBasedAlg {
+	pbs := &PriorityBasedAlg{
+		PriorityRatios: ratios,
+		taskModCounter: 1,
+	}
+
+	return pbs
+}
+
+/*
+This is the entry point to the scheduling algorithm
+It returns the list of tasks that should be assigned to nodes as per the
+algorithm documented at
+https://docs.google.com/document/d/1MqN2oAKRHHi_k29fYyUdYfw7oDN8sii0B3yB1_Yqk34/edit
```

Comment:
we shouldn't link to private docs from open source

Reply:
done